airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/uploader/BigQueryDirectUploader.java

```diff
@@ -4,29 +4,205 @@
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BigQueryDirectUploader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BigQueryDirectUploader.class);
+
+  protected final TableId table;
+  protected final WriteDisposition syncMode;
+  protected final BigQueryTableWriter writer;
+  protected final BigQuery bigQuery;
+  protected final BigQueryRecordFormatter recordFormatter;
+
+  BigQueryDirectUploader(final TableId table,
+                         final BigQueryTableWriter writer,
+                         final WriteDisposition syncMode,
+                         final BigQuery bigQuery,
+                         final BigQueryRecordFormatter recordFormatter) {
+    this.table = table;
+    this.writer = writer;
+    this.syncMode = syncMode;
+    this.bigQuery = bigQuery;
+    this.recordFormatter = recordFormatter;
+  }
+
+  public void upload(final PartialAirbyteMessage airbyteMessage) {
+    try {
+      writer.write(recordFormatter.formatRecord(airbyteMessage));
+    } catch (final IOException | RuntimeException e) {
+      LOGGER.error("Got an error while writing message: {}", e.getMessage(), e);
+      LOGGER.error(String.format(
+          "Failed to process a message for job: %s",
+          writer.toString()));
+      printHeapMemoryConsumption();
+      throw new RuntimeException(e);
+    }
+  }
+
 
-public class BigQueryDirectUploader extends AbstractBigQueryUploader<BigQueryTableWriter> {
+  public void closeAfterPush() {
+    try {
+      this.writer.close();
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public void createRawTable() {
+    // Ensure that this table exists.
+    // TODO alter an existing raw table?
+    final Table rawTable = bigQuery.getTable(table);
+    if (rawTable == null) {
+      LOGGER.info("Creating raw table {}.", table);
+      bigQuery.create(TableInfo.newBuilder(table, StandardTableDefinition.of(recordFormatter.getBigQuerySchema())).build());
+    } else {
+      LOGGER.info("Found raw table {}.", rawTable.getTableId());
+    }
+  }
+
+  /**
+   * Creates a partitioned table if the table previously was not partitioned
+   *
+   * <p>
+   * Note: this logic is deprecated since it was used for the functionality of migrating unpartitioned
+   * tables to partitioned tables for performance. Since this change was introduced in Oct 2021 there
+   * is a well founded belief that any customer's that would have ran a sync in between end of 2022
+   * and Oct 2021 would have migrated to a partition table
+   * </p>
+   *
+   * @param bigQuery BigQuery interface
+   * @param schema Schema of the data table
+   * @param destinationTableId identifier for a table
+   */
+  @Deprecated
+  public static void partitionIfUnpartitioned(final BigQuery bigQuery, final Schema schema, final TableId destinationTableId) {
```

Comment:
This can be deleted too I think. No usages and comment says end of 2022

Reply:
dv2 recreated all the raw tables anyway :shrug: so I vote to delete this

Reply:
done.